python/flight-crew-scheduling/src/flight_crew_scheduling/demo_data.py

```diff
@@ -0,0 +1,255 @@
+                departure_airport=next_flight.arrival_airport,
+                arrival_airport=pick_random_airport(remaining_airports, next_flight.arrival_airport.code)
+            )
+            flights.append(next_flight)
+            count_flights += 1
+
+        flights.append(Flight(flight_number=str(count_flights), departure_airport=next_flight.arrival_airport, arrival_airport=home_flight.departure_airport))
+        count_flights += 1
+
+    # Assign flight numbers
+    for i, flight in enumerate(flights):
+        flight.flight_number = f"Flight {i + 1}"
+
+    # Assign flight durations
+    count_dates = size // len(dates)
+
+    def flight_consumer(f: Flight, d: date):
+        key = f"{f.departure_airport.code}-{f.arrival_airport.code}"
+        count_hours = distances[key]
+        start_time = random.choice(time_groups)
+        departure_datetime = datetime.combine(d, start_time)
+
+        if departure_datetime < start_datetime:
+            departure_datetime = start_datetime + timedelta(hours=random.randint(0, 4))
+
+        arrival_datetime = departure_datetime + timedelta(hours=count_hours)
+        f.departure_utc_date_time = departure_datetime
+        f.arrival_utc_date_time = arrival_datetime
+
+    apply_random_value_with_param(count_dates, flights, random.choice(dates),
+                       lambda f: f.departure_utc_date_time is None, flight_consumer)
+
+    # Ensure no flights are left without assigned dates
+    for flight in flights:
+        if flight.departure_utc_date_time is None:
+            flight_consumer(flight, random.choice(dates))
+
+    return flights
+
+
+def generate_employees(flights: List[Flight], dates: List[date]) -> List[Employee]:
+    name_supplier = lambda: f"{random.choice(FIRST_NAMES)} {random.choice(LAST_NAMES)}"
+
+    # Get distinct departure airports from flights
+    flight_airports = list({flight.departure_airport for flight in flights})
+
+    # Two pilots and three attendants per airport
+    employees = []
+    counter = 0
+
+    for airport in flight_airports:
+        for _ in range(2):  # Two teams per airport
+            employees.append(Employee(id=str(counter := counter + 1), name=name_supplier(), home_airport=airport, skills=[PILOT_SKILL], unavailable_days=[]))
+            employees.append(Employee(id=str(counter := counter + 1), name=name_supplier(), home_airport=airport, skills=[PILOT_SKILL], unavailable_days=[]))
+            employees.append(Employee(id=str(counter := counter + 1), name=name_supplier(), home_airport=airport, skills=[ATTENDANT_SKILL], unavailable_days=[]))
+            employees.append(Employee(id=str(counter := counter + 1), name=name_supplier(), home_airport=airport, skills=[ATTENDANT_SKILL], unavailable_days=[]))
+
+            if airport.code == "CNF":
+                employees.append(Employee(id=str(counter := counter + 1), name=name_supplier(), home_airport=airport, skills=[ATTENDANT_SKILL], unavailable_days=[]))
```

Comment:
The walrus  `x := y` operator can be confusing and not usually encountered. I recommend importing [itertools.count](https://docs.python.org/3/library/itertools.html#itertools.count) and using it as an id generator
```suggestion
    ids = map(str, itertools.count())
    for airport in flight_airports:
        for _ in range(2):  # Two teams per airport
            employees.append(Employee(id=next(ids) name=name_supplier(), home_airport=airport, skills=[PILOT_SKILL], unavailable_days=[]))
            employees.append(Employee(id=next(ids), name=name_supplier(), home_airport=airport, skills=[PILOT_SKILL], unavailable_days=[]))
            employees.append(Employee(id=next(ids), name=name_supplier(), home_airport=airport, skills=[ATTENDANT_SKILL], unavailable_days=[]))
            employees.append(Employee(id=next(ids), name=name_supplier(), home_airport=airport, skills=[ATTENDANT_SKILL], unavailable_days=[]))

            if airport.code == "CNF":
                employees.append(Employee(id=next(ids), name=name_supplier(), home_airport=airport, skills=[ATTENDANT_SKILL], unavailable_days=[]))
```